core/api/rpc/client_setup.hpp

```diff
@@ -48,14 +52,15 @@ namespace fc::api::rpc {
         cb(chan);
       } else {
         c.call(std::move(req), [&c, cb{std::move(cb)}](auto &&_result) {
-          boost::asio::post(
-              c.io2, [_result{std::move(_result)}, result_cb{std::move(cb)}] {
-                if (_result) {
-                  result_cb(api::decode<Result>(_result.value()));
-                } else {
-                  result_cb(_result.error());
-                }
-              });
+          boost::asio::post(c.io2,
+                            [_result{std::forward<decltype(_result)>(_result)},
+                             result_cb{std::move(cb)}] {
+                              if (_result) {
+                                result_cb(api::decode<Result>(_result.value()));
+                              } else {
+                                result_cb(_result.error());
+                              }
```

Comment:
```suggestion
                             cb{std::move(cb)}] {
                             OUTCOME_CB(auto result, _result);
                             cb(api::decode<Result>(_result.value()));
```

Reply:
ok, but `OUTCOME_CB(auto &&result, _result);` since constructor is private.